src/Lachain.Core/RPC/HTTP/Web3/TransactionServiceWeb3.cs

```diff
@@ -594,7 +597,8 @@ private string SignTransaction(JObject opts)
                
             }
 
-            return _transactionSigner.Sign(tx, keyPair);
+            return _transactionSigner.Sign(tx, keyPair, 
+                HardforkHeights.IsHardfork_8Active(_stateManager.LastApprovedSnapshot.Blocks.GetTotalBlockHeight()));
```

Comment:
If this tx is later added to the pool, then we should use GetTotalBlockHeight() + 1

Reply:
fixed